pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -731,6 +731,21 @@ static void validateUpsertAndDedupConfig(TableConfig tableConfig, Schema schema)
             "The delete record column must be a single-valued BOOLEAN column");
       }
 
+      double deletedKeysTTL = upsertConfig.getDeletedKeysTTL();
```

Comment:
Shall we move this along with `metadataTTL` validation to avoid duplicate code on `comparisonColumns` check?